src/Apps/Legion/Hooks/useCustomNavigation.tsx

```diff
@@ -27,7 +27,15 @@ export const useCustomNavigation = (
                     : { onClick: onClick, disabled: disabled }
             }
         });
-    }, [disabled, onClick, primaryAction, wizardNavigationContextDispatch]);
+        // Stringify avoids infinite loop
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do we have to disable the rule? Aren't we using primaryAction? or does it not see that?

Reply:
I think it gets confused because it's stringified. When I remove the two eslint disable lines, I get the yellow squiggle until I add both (or add primaryAction directly, but that was causing an infinite rerender)

Reply:
I would have expected you to only need one, similar to the function below, but if you need them both leave them there.